README.md

```diff
@@ -30,40 +30,115 @@ This repository contains a tool for updating and reviewing GitHub pull requests
+
+Here's an example of what the workflow file could look like:
+
+```yaml
+name: GPT Pull Request Updater
+
+on:
+   pull_request:
+      types:
+         - opened
+         - synchronize
+
+jobs:
+   update_pull_request:
+      runs-on: ubuntu-latest
+
+      steps:
+         - name: Set up Go
+           uses: actions/setup-go@v2
+           with:
+              go-version: 1.19
+
+         - name: Checkout GPT-PullRequest-Updater
+           uses: actions/checkout@v2
+           with:
+              repository: ravilushqa/gpt-pullrequest-updater
+              path: gpt-pullrequest-updater
+
+         - name: Build description and review commands
+           run: |
+              cd gpt-pullrequest-updater
+              go build -o description ./cmd/description
+              go build -o review ./cmd/review
+
+         - name: Update Pull Request Description
+           run: |
+              ./gpt-pullrequest-updater/description
+           env:
+              GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+              OPENAI_TOKEN: ${{ secrets.OPENAI_TOKEN }}
+              OWNER: ${{ github.repository_owner }}
+              REPO: ${{ github.event.repository.name }}
+              PR_NUMBER: ${{ github.event.number }}
+
+         - name: Review Pull Request
+           if: github.event.action == 'opened'
+           run: |
+              ./gpt-pullrequest-updater/review
+           env:
+              GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+              OPENAI_TOKEN: ${{ secrets.OPENAI_TOKEN }}
+              OWNER: ${{ github.repository_owner }}
+              REPO: ${{ github.event.repository.name }}
+              PR_NUMBER: ${{ github.event.number }}
 
-```
-./review
 ```
 
-### Test Mode
+Make sure to add your OpenAI API token to your repository secrets as `OPENAI_TOKEN`.
 
-Both commands support a test mode that prints the generated content to the console instead of updating the pull request. To enable test mode, set the `TEST` environment variable to `true`:
+### Granting Permissions for GitHub Actions
 
-```
-export TEST=true
-```
+In order to use this GitHub Action, you need to grant the necessary permissions to the GitHub token. To do this, follow these steps:
 
-Then, run the desired command as described above. The generated content will be printed to the console.
+1. Go to the repository settings page: https://github.com/OWNER/REPO/settings
+2. Navigate to the "Actions" tab on the left side of the settings page.
+3. Scroll down to the "Workflow Permissions" section.
 
-## License
+Select "Read and Write" permissions for the actions. This will provide your token with the necessary rights to modify your repository.
+By following these steps, you'll grant the required permissions for the GPT-PullRequest-Updater GitHub Action to function properly, allowing it to update and review pull requests in your repository.
 
+License
 This project is licensed under the MIT License.
\ No newline at end of file
```

Comment:
The code patch adds a tool for updating and reviewing GitHub pull requests. The commands `review` and `description` are used to review pull requests and update the pull request description, respectively. The code also includes a GitHub action that can automatically run the `review` and `description` commands when a pull request is opened or synchronized. 

The `review` and `description` commands require environment variables to be set, and the GitHub action uses secrets to access the necessary tokens. Test mode can be enabled by adding the `--test` flag. 

Improvement suggestions:

- Provide more detailed documentation on how to set the necessary environment variables and secrets.
- Add error handling and logging to provide more informative error messages when there are issues.
- Consider using a more descriptive name for the GitHub action workflow, to make it clear what it does.